src/Maps/MapWriter.cpp

```diff
@@ -74,8 +74,13 @@ namespace MapWriter {
 		{
 			WriteContainerSize(streamWriter, tileGroups.size());
 
+			uint32_t unknown = 0;
+
 			// tileGroups.size is checked to ensure it is below UINT32_MAX by previous call to WriteContainerSize.
-			uint32_t unknown = static_cast<uint32_t>(tileGroups.size()) - 1; // Write unknown field with best guess as to what value it should hold
+			if (!tileGroups.empty()) {
+				unknown = static_cast<uint32_t>(tileGroups.size()) - 1; // Write unknown field with best guess as to what value it should hold
+			}
+			
```

Comment:
This looks good. Could possible make it more compact using the ternary operator:
```cpp
// Write unknown field with best guess as to what value it should hold
uint32_t unknown = !tileGroups.empty() ? static_cast<uint32_t>(tileGroups.size()) - 1 : 0;
streamWriter.Write(unknown);
```

I'd also suggest fewer blank lines around the `unknown` field code, so it groups together visually.